py-polars/tests/unit/series/test_series.py

```diff
@@ -2296,10 +2296,19 @@ def test_product() -> None:
     assert out == 6
     a = pl.Series("a", [1, 2, None])
     out = a.product()
-    assert out is None
+    assert out == 2
```

Comment:
Ignoring `null` values by default IS a breaking change. I'd suggest leaving that for a different PR, and making this one just add the identity for empty Series.

Reply:
What about `sum`? It currently doesn't have such an option and already ignores nulls. I'm just making it consistent with `sum`.

Reply:
I get that, but it would be a breaking change. Ignoring nulls is not a bug.

Reply:
I would say our product implementation is a bug. It didn't adhere to our set semantics/rules. Given that 0.19.0 will be patched and yanked I will include this one.